crates/sui-core/src/consensus_handler.rs

```diff
@@ -829,6 +858,104 @@ impl ConsensusCommitInfo {
+    /// Whether to enable handling certified transactions.
+    enabled: bool,
+    /// Per-epoch store.
+    epoch_store: Arc<AuthorityPerEpochStore>,
+    /// Enqueues transactions to the transaction manager via a separate task.
+    transaction_manager_sender: TransactionManagerSender,
+    /// Metrics for consensus transaction handling.
+    metrics: Arc<AuthorityMetrics>,
+}
+
+impl ConsensusTransactionHandler {
+    pub fn new(
+        epoch_store: Arc<AuthorityPerEpochStore>,
+        transaction_manager_sender: TransactionManagerSender,
+        metrics: Arc<AuthorityMetrics>,
+    ) -> Self {
+        Self {
+            enabled: epoch_store.protocol_config().mysticeti_fastpath(),
+            epoch_store,
+            transaction_manager_sender,
+            metrics,
+        }
+    }
+
+    pub fn enabled(&self) -> bool {
+        self.enabled
+    }
+
+    pub async fn handle_consensus_transactions(&self, parsed_transactions: Vec<ParsedTransaction>) {
+        let mut pending_consensus_transactions = vec![];
+        let executable_transactions: Vec<_> = parsed_transactions
+            .into_iter()
+            .filter_map(|parsed| {
+                // TODO(fastpath): unlock rejected transactions.
+                // TODO(fastpath): maybe avoid parsing blocks twice between commit and transaction handling?
+                if parsed.rejected {
+                    self.metrics
+                        .consensus_transaction_handler_processed
+                        .with_label_values(&["rejected"])
+                        .inc();
+                    return None;
+                }
+                self.metrics
+                    .consensus_transaction_handler_processed
+                    .with_label_values(&["certified"])
+                    .inc();
+                match &parsed.transaction.kind {
+                    ConsensusTransactionKind::UserTransaction(tx) => {
+                        if tx.contains_shared_object() {
+                            return None;
+                        }
+                        pending_consensus_transactions.push(parsed.transaction.clone());
+                        let tx = VerifiedTransaction::new_from_verified(*tx.clone());
+                        Some(VerifiedExecutableTransaction::new_from_consensus(
+                            tx,
+                            self.epoch_store.epoch(),
+                            parsed.round,
+                            parsed.authority,
+                            parsed.transaction_index,
+                        ))
+                    }
+                    _ => None,
+                }
+            })
+            .collect();
+
+        if pending_consensus_transactions.is_empty() {
+            return;
+        }
+        {
+            let reconfig_state = self.epoch_store.get_reconfig_state_read_lock_guard();
+            // Stop executing fastpath transactions when epoch change starts.
```

Comment:
For my own edification, as long as we hold the `reconfig_state` lock it is impossible for epoch change to happen correct? Just want to make sure that the `self.epoch_store.insert_pending_consensus_transactions` won't fail due to an underlying epoch change.